### PR TITLE
[CIS-474] Presenting Suggestions on ChatVC

### PR DESCRIPTION
**Pull request by DominikBucher12**
# What this PR does
- Presents SuggestionsViewController on ChatVC as a child VIewController, this eliminates problems with strange behaviour while adding it as VC on custom Window or child of `InputAccessoryViewController`. Also resizes and moves according to position of inputView. 
# How to test this
- Go to chat and run suggestions either with the flash button or with typing `/`, see if this works for you

### CODE REVIEW

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift [hunk 1]
```diff
@@ -81,25 +77,53 @@ open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>:
...
     override public func setUpLayout() {
-        view.translatesAutoresizingMaskIntoConstraints = false
-
-        heightConstraint = collectionView.heightAnchor.constraint(equalToConstant: 0)
+        view.embed(containerView)
+        containerView.addSubview(collectionView)
+        collectionView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
+        collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        collectionView.leadingAnchor.constraint(equalTo: containerView.layoutMarginsGuide.leadingAnchor).isActive = true
+        collectionView.trailingAnchor.constraint(equalTo: containerView.layoutMarginsGuide.trailingAnchor).isActive = true
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift:93]: You can use the `pin` helpers

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift [hunk 2]
```diff
@@ -9,17 +9,18 @@ open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>:
...
     public enum State {
         case commands([Command])
         case mentions([String])
     }
 
     // MARK: - Property
-    
+
+    private var frameObserver: NSKeyValueObservation?
+    public var bottomAnchorView: UIView?
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift:30]: tip: You don't need to be afraid of putting comments to your code such that others can understand what you mean. What is `bottomAnchorView ` and why we need it?

File: Sources_v3/StreamChatUI/ChatChannel/ChatVC.swift [hunk 3]
```diff
@@ -16,14 +16,18 @@ open class ChatVC<ExtraData: ExtraDataTypes>: ViewController,
 
     public var channelController: _ChatChannelController<ExtraData>!
 
+    public private(set) lazy var suggestionsViewController: MessageComposerSuggestionsViewController<ExtraData> = {
+        uiConfig.messageComposer.suggestionsViewController.init()
+    }()
+
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/ChatVC.swift:22]: niL: The `{` ...` }()` is redundant

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift [hunk 4]
```diff
@@ -5,23 +5,31 @@
...
 
+public protocol SuggestionsViewControllerPresenter: class {
+    func showSuggestionsViewController(
+        with state: SuggestionsViewControllerState,
+        onSelectItem: ((Int) -> Void)
+    )
+    func dismissSuggestionsViewController()
+
+    var isSuggestionControllerPresented: Bool { get }
+}
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift:16]: Why do we need this abstraction?

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift [hunk 5]
```diff
@@ -5,23 +5,31 @@
...
+
+public enum SuggestionsViewControllerState {
+    case commands([Command])
+    case mentions([String])
+}
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift:21]: Are you sure about this? How will this connect to `UserSearchController`?

File: Sources_v3/StreamChatUI/ChatChannel/ChatVC.swift [hunk 6]
```diff
@@ -157,3 +161,24 @@ open class ChatVC<ExtraData: ExtraDataTypes>: ViewController,
...
+extension ChatVC: SuggestionsViewControllerPresenter {
+    public var isSuggestionControllerPresented: Bool {
+        suggestionsViewController.isPresented
+    }
+
+    public func showSuggestionsViewController(with state: SuggestionsViewControllerState, onSelectItem: ((Int) -> Void)) {
+        suggestionsViewController.state = state
+        addChild(suggestionsViewController)
+        view.addSubview(suggestionsViewController.view)
+        suggestionsViewController.didMove(toParent: parent)
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/ChatChannel/ChatVC.swift:176]: You can/should use our helper for this

**Review by VojtaStavik — COMMENTED** [review]
Let's have a call and talk about the details.

File: Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift [hunk 7]
```diff
@@ -42,11 +43,18 @@ open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>:
         .init(layout: uiConfig.messageComposer.suggestionsCollectionViewLayout.init())
         .withoutAutoresizingMaskConstraints
 
+    open private(set) lazy var containerView: UIView = UIView().withoutAutoresizingMaskConstraints
+
```
[the remark this answers is not in the document]
Comment (DominikBucher12) [reply Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift:55]: When you pin the collectionView just to the layoutMargins of the RootViewController of the child, the constants are 0 and not the standart 8. This container ensures that the margins are right.

**Review by VojtaStavik — APPROVED** [review]
OK, let's merge this as is and focus on the data source changes we talked about IRL.

**Pull request merged**